Use the correct executable for debugging

danielrainer opened the pull request:
The previous version of the code just uses the first executable it finds in the JSON output of cargo. This is problematic because multiple executables might be produced, for example when debugging an integration test, and the order of these artifacts in the JSON output can change from one invocation of cargo to the next.

Analyze the arguments given to cargo instead, from which the expected values for target.kind and target.name can be extracted. With this information, the executable should be uniquely identifiable.

I know very little about TypeScript so feel free to point out/improve any issues.

fannheyward commented:
Thanks for your PR, will check this ASAP.

fannheyward commented:
> when debugging an integration test, and the order of these artifacts in the JSON output can change from one invocation of cargo to the next

Can you give an example code for this case?

danielrainer commented:
I fixed the linter errors and added some example code here: https://github.com/danielrainer/coc-rust-analyzer-debug-repro

Try to run the functions provided there with `:CocCommand rust-analyzer.debug`. With the changes from this PR applied, the correct functions should always run. Without them, the wrong executable name might be selected, so the debugger runs something other than the function which should run.

Because the current filename parsing reads cargo's JSON output which is not always in the same order, the behavior can be different from one invocation of the debugger to the next.
